r/tests/testthat/test-s3-minio.R

```diff
@@ -61,6 +61,13 @@ fs$CreateDir(now)
 # Clean up when we're all done
 withr::defer(fs$DeleteDir(now))
 
+test_that("Confirm s3_bucket works with endpoint_override", {
+  
+  
+  bucket <- s3_bucket("test", endpoint_override = "https://play.min.io")
```

Comment:
@paleolimbot good question, that's something I'm trying to figure out too.  It failed to find the bucket in the previous test, https://github.com/apache/arrow/actions/runs/4093029707/jobs/7073398005, but I don't think that's because of any change I made -- in my own local testing I could never reproduce that error.  I see the desire to use localhost, but it's not quite clear how to ensure the localhost deploy works before running the test.

It's not so clear to me that any of the other tests rely on the localhost bucket existing?  

https://play.min.io is the official test service for the MINIO platform, not sure if that's ideal choice here for the tests, but then the test suite here isn't really built out and my objective is just submit a patch for this confirmed bug in the codebase.